library/cwm/src/lib/cwm/table.rb

```diff
@@ -0,0 +1,119 @@
+require "abstract_method"
+require "cwm/custom_widget"
+
+module CWM
+  # Represents Table widget
+  class Table < CustomWidget
+    # @method header
+    #   return array of String or Yast::Term which is used as headers for table
+    #   it can use e.g. align Left/Center/Right
+    # @note It has to be overwritten
+    # @return [Array<String|Yast::Term>]
+    #
+    # @example of header
+    #   def header
+    #     [
+    #       _("First Name"),
+    #       Right(_("Surname")
+    #     ]
+    #   end
+    abstract_method :header
+
+    # gets initial two dimensional array of Table content
+    # one element in the first dimension contain as first element id and then
+    # rest is data in table, which can be e.g. terms. Then it have to be enclosed in
+    # `cell` term.
+    # @see for more complex example see examples directory
+    # @note default value is empty array. It is useful when computation expensive content
+    #   need to be set. In such case, it is better to keep empty items to quickly show table and
+    #   then in #init call #change_items method, so it will be filled when all widgets are at place
+    #   and just filling its content.
+    #
+    # @example for table with two collumns
+    #   def items
+    #     [
+    #       [:first_user, "Joe", "Doe"],
+    #       [:best_user, "Chuck", "Norris"]
+    #     ]
+    #   end
+    def items
+      []
+    end
+
+    # change list on fly with argument. Useful when content of widget is changed.
+    # @param items_list [Array<Array<Object>>] same format as {#items}
+    def change_items(items_list)
```

Comment:
I am still not sure how to handle the distinction between the _initial_ `items` and the _current_ `change_items`.
... Ah! it is in fact consistent with the [ItemsSelection](http://www.rubydoc.info/github/yast/yast-yast2/CWM/ItemsSelection) mixin. Let's mention that name somewhere.